src/class_singleListener.py

```diff
@@ -86,12 +86,13 @@ def run(self):
         # we'll fall back to IPv4-only.
         try:
             sock = self._createListenSocket(socket.AF_INET6)
-        except socket.error, e:
+        except socket.error as e:
             if (isinstance(e.args, tuple) and
                 e.args[0] in (errno.EAFNOSUPPORT,
                               errno.EPFNOSUPPORT,
                               errno.EADDRNOTAVAIL,
-                              errno.ENOPROTOOPT)):
+                              errno.ENOPROTOOPT,
+                              'getsockaddrarg: bad family')):
```

Comment:
This article https://docs.python.org/2/library/socket.html#socket.error says socket.error may return string instead of errno and i've seen it.